Source/Scene/Model.js

```diff
@@ -2058,8 +2069,13 @@ define([
                 pickFS = modifyShaderForClippingPlanes(pickFS, clippingPlaneCollection);
             }
 
-            pickVS = ModelUtility.modifyVertexShaderForLogDepth(pickVS, toClipCoordinatesGLSL);
-            pickFS = ModelUtility.modifyFragmentShaderForLogDepth(pickFS);
+            // Internet Explorer seems to have problems with discard after too many levels of indirection:
```

Comment:
I would move this comment to the first check instead of last so readers encounter it sooner.